getLocationsInArea function added

BBrunekreeft opened the pull request:
Retrieves PostNL Locations in specific area.
Code Example:

```
$coordinatesNorthWest = \DivideBV\Postnl\ComplexTypes\Coordinate::create()
    ->setLongitude(5.085307)
    ->setLatitude(52.118860);
$coordinatesSouthEast = \DivideBV\Postnl\ComplexTypes\Coordinate::create()
    ->setLongitude(5.135604)
    ->setLatitude(52.079527);
$locations = $postNL->getLocationsInArea(
    $coordinatesNorthWest,
    $coordinatesSouthEast,
    'false',
    null,
    ['PG', 'PGE'],
    ['Daytime'],
    'NL');
```

BBrunekreeft committed:
Update LocationClient.php

Added getLocationsInArea method

BBrunekreeft committed:
Update Postnl.php

ameenross commented on src/Postnl.php:
```diff
@@ -417,6 +417,35 @@ public function getNearestLocation(
+     * @return ComplexTypes\GetNearestLocationsResponse
+     */
+    public function getLocationsInArea(
+        $coordinatesNorthWest,
```
Please use type hints

ameenross commented on src/Postnl.php:
```diff
@@ -417,6 +417,35 @@ public function getNearestLocation(
+        $locationArea = new ComplexTypes\LocationArea($allowSundaySorting, $deliveryDate, $deliveryOptions, $options);
+        $locationArea
+                ->setCoordinatesNorthWest($coordinatesNorthWest)
+                ->setCoordinatesSouthEast($coordinatesSouthEast);
```
If you use method chaining, I prefer you use the factory, I.E.

```php
$locationArea = ComplexTypes\LocationArea::create()
    ->setAllowSundaySorting($allowSundaySorting)
    ->setDeliveryDate($deliveryDate)
    //etc
```

ameenross reviewed (commented):
Thx for the contribution. Please have a look at [CONTRIBUTING.md](https://github.com/DivideBV/Postnl/blob/master/CONTRIBUTING.md). Check your code style. Also, I'd like the commits to be a bit more descriptive.

BBrunekreeft committed:
Type hints

Use type hints for parameters $coordinatesNorthWest and $coordinatesSouthEast in function getLocationsInArea()

BBrunekreeft committed:
Use factory with method chaining

Use ComplexTypes\LocationArea::create() with method chaining in function getLocationsInArea.
Give default value to $deliveryDate if deliverydate is null

slokhorst commented:
I would like to use this function. Is there any more work needed?

BBrunekreeft commented:
No more work needed

ameenross commented on src/Postnl.php:
```diff
@@ -417,6 +417,40 @@ public function getNearestLocation(
+    public function getLocationsInArea(
+        ComplexTypes\Coordinate $coordinatesNorthWest,
+        ComplexTypes\Coordinate $coordinatesSouthEast,
+        $allowSundaySorting = 'false',
```
Does this need to be a string? Or can it be a boolean

BBrunekreeft replied:
This is what the documentations says:

> AllowSundaySorting:  String [M]  - Possible values are true/false (no capitals allowed)

So I think it should be a string.

ameenross commented:
Sorry for the delay, I'd like to get this merged soon
